aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1182,10 +1182,12 @@ static Tensor squeeze_multiple(const Tensor& self, IntArrayRef dims) {
 static Tensor& logsumexp_out_impl(Tensor& result, const Tensor& self, IntArrayRef dims, bool keepdim) {
   // can't take max of empty tensor
   if (self.numel() != 0) {
-    auto maxes = at::amax(self, dims, true);
+    Tensor self_;
```

Comment:
Since this is an internal function I would just have a contract that the result tensor must have the correct dtype, so this function does not need to change. Instead, add a `TORCH_CHECK` to `logsumexp_out` to check that the result tensor is not integral dtype (including bool) as we should not be modifying the result tensor passed in by the user. As for `logsumexp` we can promote integral inputs to the default floating point dtype when creating the result tensor.

Reply:
Thank so much for such detailed info, I have leave a dtype check for result tensor, and promote the integral inputs in `logsumexp`, PTAL.